controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/MetricsReporter.java

```diff
@@ -121,8 +129,37 @@ private double deploymentFailRatio() {
                                                         .asList();
         if (applications.isEmpty()) return 0;
 
-        return (double)applications.stream().filter(a -> a.deploymentJobs().hasFailures()).count() / 
-               (double)applications.size();
+        return (double) applications.stream().filter(a -> a.deploymentJobs().hasFailures()).count() /
+               (double) applications.size();
+    }
+
+    private Map<ApplicationId, Duration> averageDeploymentDurations() {
+        Instant now = clock.instant();
+        return ApplicationList.from(controller().applications().asList())
+                              .notPullRequest()
+                              .hasProductionDeployment()
+                              .asList()
+                              .stream()
+                              .collect(Collectors.toMap(Application::id,
+                                                        application -> averageDeploymentDuration(application, now)));
+    }
+
+    private Duration averageDeploymentDuration(Application application, Instant now) {
+        List<Duration> jobDurations = application.deploymentJobs().jobStatus().values().stream()
+                                                 .filter(status -> status.lastTriggered().isPresent())
+                                                 .map(status -> {
+                                                     Instant triggeredAt = status.lastTriggered().get().at();
+                                                     Instant completedAt = status.lastCompleted()
```

Comment:
"completedAt" is misleading; call this "runningUntil" or something?

Reply:
Done.